src/web/client/context/fileDataMap.ts

```diff
@@ -71,4 +71,19 @@ export class FileDataMap {
         }
         throw Error("File does not exist in the map"); // TODO - Revisit errors and dialog experience here
     }
+
+    searchFiles(pattern: string) {
+        // create case sensitive regex
+        const regex = new RegExp(pattern, "i");
+
+        const files: FileData[] = [];
+        this.fileMap.forEach((fileData) => {
```

Comment:
lets move this to fileSystemProvider this.root tree lookup instead - that will be the place for both text and file name search